proxy/healthy_endpoints.go

```diff
@@ -22,6 +23,7 @@ func (h *healthyEndpoints) filterHealthyEndpoints(endpoints []routing.LBEndpoint
 	for _, e := range endpoints {
 		if p < e.Metrics.HealthCheckDropProbability() {
 			/* drop */
+			metrics.IncCounter("routing.endpoint." + e.Host + ".optout.loadbalancer")
```

Comment:
I think it is better to use Sprintf here:
`fmt.Sprintf("routing.endpoint.%s.optout.loadbalancer", e.Host)`.

Reply:
Conactenation is fine but hostname should be escaped, see https://github.com/zalando/skipper/pull/3020

Reply:
Or maybe use https://github.com/zalando/skipper/blob/708e97af1d1e1511d7bc7803e1b2a0d5599df52b/metrics/utils.go#L21-L25

Reply:
How much memory we think this will take? IIRC int64
What if we have a non trivial number of all day crashing pods? what if we have 10k pods?
Why do we need `e.Host` in the metric?

We don't clean up metrics, so we store it for lifetime of the process. This host is not set by a client (unlike Host header for example), so maybe it's fine.

Reply:
This host is an IP of the pod so metrics size concern is very valid.

Maybe `optout -> drop` to stick to the parlance.

Reply:
> (I think its always host+port)

Yes, it is, see https://github.com/zalando/skipper/pull/2824.

Reply:
I added `e.Host` to define which IP we escape then we can track on AWS